File: frontend/src/types/address.ts
export interface AddressRequest {
    street: string;
    city: string;
    ward: string;
    district: string;
}
export interface AddressResponse extends AddressRequest {
    id: number;
    isDefault?: boolean;
    userId?: number;
    receiverName?:string;
    receiverPhone:string;
}

File: backend/src/main/java/com/tientoan21/entity/Address.java
package com.tientoan21.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "addresses")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String street;
    private String city;
    private String ward;
    private String district;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    private Boolean isDefault;
}
